data/mil2525d/utilities/military-features-utilities/template-gdb-toolbox/scripts/CreateTemplateGDB/AddFieldsFromSchema.py

```diff
@@ -69,38 +131,45 @@ def addFieldsFromSchema(schemasFolder, featureClass, schema):
                     reader = csv.reader(csvFile, dialect='excel')
             
                     # Skip the headers
+                    
                     header = next(reader)
                     
                     # Read all the rows and add fields accordingly
+                    
+                    fieldName = None
+            
                     for line in reader:
+                        fieldType = Utility.fieldTypeLookup(line[1])
                         
-                        fieldType = fieldTypeLookup(line[1])
+                        # Add a line as a new field if this is the first occurence of that "field"
                         
-                        if fieldType == "TEXT":
-                            length = int(line[2])
+                        if line[0] != fieldName:
+                            addAField(featureClass, fieldType, line)
+                            fieldName = line[0]
                             
-                            
-                            arcpy.AddField_management(featureClass, \
-                                                      field_name=line[0], \
-                                                      field_type=fieldType, \
-                                                      field_length=length, \
-                                                      field_alias=line[3], \
-                                                      field_is_nullable=line[4], \
-                                                      field_domain=line[5])
-                        else:
-                            arcpy.AddField_management(featureClass, \
-                                                      field_name=line[0], \
-                                                      field_type=fieldType, \
-                                                      field_alias=line[3], \
-                                                      field_is_nullable=line[4], \
-                                                      field_domain=line[5])
+                        # Set the domain for the field, including subtype code if necessary
                         
+                        if bool(line[5]):
```

Comment:
For all of these `line[index]`entries you should have a well-named local variable defined/assigned above for these instead - this code will be quite hard to maintain otherwise